sdk/src/org.graalvm.nativeimage/src/org/graalvm/nativeimage/hosted/RuntimeSerialization.java

```diff
@@ -55,6 +55,27 @@
 @Platforms(Platform.HOSTED_ONLY.class)
 public final class RuntimeSerialization {
 
+    /**
+     * Register the specified serialization target class itself and all associated classes.
+     * <p>
+     * According to the Java Object Serialization Specification, the associated classes include 1)
+     * all the target class' non-static and non-transient fields types and their associated classes;
+     * 2) other fields defined in the customised writeObject(ObjectOutputStream) and
+     * readObject(ObjectInputStream). This method can automatically explore all possible
+     * serialization target classes in the first scenario, but can't figure out the classes in the
+     * second scenario.
+     * <p>
+     * Another limitation is the specified {@code clazz} must have no subclasses (effectively
+     * final). Otherwise, the actual serialization target class could be any subclass of the
+     * specified class at runtime.
+     *
+     * @param clazz the serialization target class
+     * @Since 23.0
+     */
+    public static void registerAllAssociatedClasses(Class<?> clazz) {
```

Comment:
Can we call this `registerIncludingAssociatedClasses`? The name sounds like it might not register the class itself.

Reply:
'registerIncludingAssociatedClasses' is too long, what about `registerWithAssociatedClasses`?

Reply:
I don't see that as a problem.
It's exactly as long as `registerWithTargetConstructorClass` a few lines below.

Reply:
Renamed the method to registerIncludingAssociatedClasses as suggested.